src/main/java/com/hida/Id.java

```diff
@@ -0,0 +1,171 @@
+
+    public abstract String convert();
+
+    
+    /**
+     * Generated by contract because equals was overridden. Prime numbers were
+     * chosen to lower the number of collisions. Increasing them to a higher
+     * prime number will lower collisions at the cost of run time.
+     *
+     * @return
+     */
+    @Override
+    public int hashCode() {
+        // arbitrarily chosen prime numbers
+        final int prime1 = 37;
+        final int prime2 = 257;
+
+        int hash = prime1 * prime2 + Arrays.hashCode(this.BaseMap);
+        return hash;
+    }
+
+    /**
+     * Overridden so that id's can be identified solely by its baseName.
+     *
+     * @param obj - the Object this id is being compared to
+     * @return - true if the two Objects are the same.
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof Id)) {
+            return false;
+        }
+        final Id paramId = (Id) obj;
+
+        // equivalent to Arrays.hashCode(a) == Arrays.hashCode(b)
+        return Arrays.equals(this.BaseMap, paramId.BaseMap);
+    }
+
+    /**
+     * The BaseMap is the numerical representation of this Id's name. Used in
+     * conjunction with TokenMap to return the string representation of this
+     * id's name in its toString method.
+     *
+     * @return - The array used to create the name
+     */
+    public int[] getBaseMap() {
+        return BaseMap;
+    }
+
+    /**
+     * The BaseMap is the numerical representation of this Id's name. Used in
+     * conjunction with TokenMap to return the string representation of this
+     * id's name in its toString method.
+     *
+     * Be warned that the array must have a unique address for it to work with
+     * Sets. The length of the array must be equal to TokenMap, otherwise an
+     * IndexOutOfBounds error will be thrown in the convert method.
+     *
+     * @param baseMap - The new array to replace the name.
+     */
+    public void setBaseMap(int[] baseMap) {
+        this.BaseMap = baseMap;
+    }
+    
+    /**
+     * Determines whether or not particular Id is not the first to be created
+     * with it's particular BaseMap.
+     *
+     * Returns true by default unless previously modified.
+     *
+     * @return
+     */
+    public boolean isUnique() {
+        return Unique;
+    }
+
+    /**
+     * Determines whether or not particular Id is not the first to be created
+     * with it's particular BaseMap.
+     *
+     * Should only be used to set false whenever an it is determined that this
+     * Id is not the first to have it's BaseMap value.
+     *
+     * @param isUnique
+     */
+    public void setUnique(boolean isUnique) {
+        this.Unique = isUnique;
+    }
+
+    public String getPrefix() {
+        return Prefix;
+    }
+
+    /**
+     * Used to define the natural ordering of how id's should be listed. When

```

Comment:
minor: since this is an `@Override` method that fulfills the requirements of the method, it's not necessary to write a javadoc here.